packages/w3p/src/types/wallet-connect.ts

```diff
@@ -0,0 +1,5 @@
+export type WalletConnectInitArgs = {
+  projectId: string
+  currentChains: [number, ...number[]]
+  optionalChains?: [number, ...number[]]
```

Comment:
what is this? Why not just `number[]`?

Reply:
Because the wallet connect uses the `ArrayOneOrMore` type, which is not imported from it's library. This type says that you need to be sure to pass an array with at least one value, and an empty array is a valid value for `number[]`, so you have to use this notation